### DISCUSSION TITLE
Versioning scheme

### DISCUSSION THREAD

**Discussion by ptomato** [gjsify/ts-for-gir#365] [General]
What versioning scheme should this module follow? An alternating stable-unstable one like GNOME does, so that you can derive the corresponding GJS version directly from the module's version? or semver, where you bump the major version every time a new version of the module can't be used with previous GJS, and there's like a table in the docs that says "4.x ↔ GJS 1.76–82" or whatever.

**Comment by JumpLink** [comment]
Good point. It should definitely be documented and maintained somewhere. Maybe we can tag types in the comments with @version @deprecated tags and then support a wide range of GJS versions and only remove old types for major breaking changes in GJS? I'm still completely open here

**Comment by JumpLink** [comment]
@ptomato  That’s still open to discussion; I’m planning to release 4.0.0 soon, so we could agree on following Semver at this point as suggested?

**Reply by ptomato** [reply]
I think it's a difficult question and I don't have a good answer. People are probably going to want to install types from whatever GJS version is on their system, or in their Flatpak runtime, so semver might not be the easiest.

**Reply by JumpLink** [reply]
Yes, in such cases it might be best to create the types locally and import them rather than using those published on NPM or we could introduce meta-packages for Flatpak runtimes, which would then be a package that includes all the types in the correct version as dependencies
